spacy/tests/parser/test_parse.py

```diff
@@ -451,10 +512,9 @@ def test_distill():
 @pytest.mark.parametrize(
     "parser_config",
     [
-        # TransitionBasedParser V1
-        ({"@architectures": "spacy.TransitionBasedParser.v1", "tok2vec": DEFAULT_TOK2VEC_MODEL, "state_type": "parser", "extra_state_tokens": False, "hidden_width": 64, "maxout_pieces": 2, "use_upper": True}),
```

Comment:
Did we make a decision on this - I thought we could just move the appropriate code to legacy and keep v1 as is?

Reply:
The test in legacy itself is not exactly the same thing though right? Having the test also in spaCy ensures that the architecture is in fact available (from legacy) and behaves similar as the other architectures (i.e. is a drop-in). Or am I missing something?

Reply:
Ah, I see! I will try to add it back to the test.

Reply:
Thanks! Basically the legacy/compat discussion is the main thing I wanted to understand fully before merging this in :-)

Reply:
I added back the test, but commented it out. What I think we need to do:

- Merge the refactor into v4.
- Do a v4 dev release.
- Create a `spacy-legacy` branch for v4.
- This PR against `spacy-legacy`: https://github.com/explosion/spacy-legacy/pull/36
- Do a `spacy-legacy` release.
- Update `v4` to require this legacy release and uncomment the test.

Tested locally and with the `spacy-legacy` PR mentioned above the test passes.

Reply:
Fun! 😁 

Thanks for spelling this out Daniël, that sounds like a perfect plan!

Reply:
I'll convert it into a ticket, so that we don't forget to do it.